extensions-api/src/main/scala/pl/touk/nussknacker/engine/api/deployment/CustomAction.scala

```diff
@@ -32,3 +33,8 @@ case class CustomActionParameter(name: String, editor: ParameterEditor)
 case class CustomActionRequest(name: String, processVersion: ProcessVersion, user: User, params: Map[String, String])
 
 case class CustomActionResult(req: CustomActionRequest, msg: String)
+
+sealed trait CustomActionDisplayPolicy
+
+// TODO: Add more
+case object CurrentlyViewedProcessVersionIsDeployed extends CustomActionDisplayPolicy
```

Comment:
I'm not sure if it is good approach. Those policies seem to be instance-specific, but each NU instance can be configured with different custom actions, different policies. 

What happens when I want to add my own custom policy? E.g I want a button that is enabled when scenario diagram contains specified node. Do I need to hardcode it here and in designer/client/src/helpers/customActionDisplayabilityResolver.ts ?

Reply:
Answering to your very last question, the answer is yes. If such hardcoded solution cannot be taken into account, the natural improvement is to create some configuration language (implemented using yaml, json, whatever), send it thru' the backend and teach frontend to read it. And of course we should not then use selaed traits.